cmd/helm/root.go

```diff
@@ -155,7 +158,13 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 
 	registryClient, err := newDefaultRegistryClient(false)
 	if err != nil {
-		return nil, err
+		// see issues #11009
+		debug("generate registry client failed: %v", err.Error())
+		for i := 0; len(args) > 0 && i < len(CMDNeedRegistryClient); i++ {
+			if CMDNeedRegistryClient[i] == args[0] {
```

Comment:
I don't think `args[0]` will work here. As e.g. `helm --debug version` (I think, should confirm) will have `arg[0] == "--debug"`. Perhaps `cmd.Name()` or `cmd.CalledAs()` ?